src/core/border_router/routing_manager.cpp

```diff
@@ -740,6 +753,54 @@ void RoutingManager::HandleLocalOnLinkPrefixChanged(void)
     return;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DISTINCT_AIL_PREFIX_ENABLE
+void RoutingManager::UpdateDistinctAilPrefixPublication(void)
+{
+    // This method is called when MultiAilDetector state changes or when
+    // the favored on-link prefix on the AIL changes.
+
+    Ip6::Prefix ailPrefix = mOnLinkPrefixManager.GetFavoredPrefix();
+    bool shouldPublish    = mMultiAilDetector.IsDetected() && ailPrefix.IsValid() && (ailPrefix != mPublishedAilPrefix);
+
+    // If we are currently publishing a prefix, check if we should withdraw it.
+    if (mIsDistinctAilPrefixPublished)
+    {
+        if (!shouldPublish || (mPublishedAilPrefix != ailPrefix))
+        {
+            // Withdraw the old prefix if we no longer should publish,
+            // or if the AIL prefix itself has changed.
+            IgnoreError(Get<NetworkData::Publisher>().UnpublishPrefix(mPublishedAilPrefix));
+            mIsDistinctAilPrefixPublished = false;
+        }
+    }
+
+    // If we should publish a prefix, check if we can.
+    if (shouldPublish && !mIsDistinctAilPrefixPublished)
+    {
+        NetworkData::OnMeshPrefixConfig config;
+
+        config.Clear();
+        config.mPrefix       = ailPrefix;
+        config.mStable       = true;
+        config.mSlaac        = true;
+        config.mPreferred    = true;
+        config.mOnMesh       = true;
+        config.mDefaultRoute = false;
+
```

Comment:
Another quick comment. This code here does not look correct. You are publishing the on-link prefix also as an "on-mesh" prefix within the mesh. So the same prefix is both on-link (on AIL/WiFi) and on-mesh, which does not make sense.

My guess is that you intended to have it published as an explicit off-mesh route in the network data, not an on-mesh prefix?